code/smartPointers/smartPointers.cpp

```diff
@@ -30,21 +31,21 @@
  */
 
 // Declare a function to do something with the data. No need to change it.
-double sumEntries(const double* data, std::size_t size) {
-    if (size > 200)
-        throw std::invalid_argument("I only want to sum 200 numbers or less.");
+template<typename T>
+double sumEntries(const T * data) {
```

Comment:
I was sure there is an error when you call `data->begin()` later until I realized you pass a `std::array<T>*`. Consider renaming the template paramter to make this clearer:
```suggestion
template<typename Range>
double sumEntries(const Range * data) {
```

Reply:
> I was sure there is an error when you call data->begin()

I compile before I commit. 🙂
But it's a good idea to call it range, so we'll go for that!